datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/validators/validator.py

```diff
@@ -200,6 +200,55 @@ def extract_oid_from_symbol(self, metric, file):
             self.duplicated[OID] = self.used_oid[OID]
 
 
+class SysobjectidValidator(ProfileValidator):
+    """
+    Validator responsible to check if there are no duplicated sysobjectid in the profile.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.used_sysobjid = {}
+        self.duplicated = {}
+        self.seen_profiles = {}
+
+    def validate(self, profile, path):
+        sysobjectids = self.extract_sysobjectids_profile(profile)
+        self.check_sysobjectids_are_duplicated(sysobjectids, profile)
+        for directory in path:
+            for profile in get_all_profiles_directory(directory):
+                sysobjectids = self.extract_sysobjectids_profile(profile)
```

Comment:
We are list all the files and extracting the sysobjectids each time we call `validate` (looks like the path doesn't change), that seems not very efficient. 

Maybe we can cache the results to avoid reprocessing the same thing too many times. 

Note that the snmp validator is quite slow compared to other validators already:
![image](https://user-images.githubusercontent.com/49917914/128385906-f8077b17-839d-4a68-a7c1-c8ee689e6d86.png)